### PR TITLE
GHA: add workflow to publish python client on PyPI

### PR DESCRIPTION
**Pull request by antisaling**
## Description

This builds and publishes tagged releases of the Python client.

The tags should be prefixed with `py-v`, then followed by the version number.

Related: #130 

## Merge criteria:

- [x] The commits and have meaningful messages; the author will squash them [after approval](https://github.com/opendatahub-io/opendatahub-community/blob/main/contributor-cheatsheet.md#:~:text=Usually%20this%20is%20done%20in%20last%20phase%20of%20a%20PR%20revision) or will ask to merge with squash.
- [ ] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [ ] The developer has manually tested the changes and verified that the changes work

### CODE REVIEW

File: .github/workflows/python-release.yml [hunk 1]
```diff
@@ -0,0 +1,47 @@
+name: Release Python client
+
+on:
+  push:
+    tags:
+      - py-v*
```
Comment (tarilabs) [inline .github/workflows/python-release.yml:6]: it's not very clear to me however what is the role of git-tags, if anyway the actual published version is identified by the toml:

https://github.com/opendatahub-io/model-registry/blob/0621864cea0d438df528afd1aac0bd6eb39fc0c2/clients/python/pyproject.toml#L3

**Review by tarilabs — COMMENTED** [review]
I believe workflow_dispatch is very meaningful so we can manually execute the publish any time required!

File: .github/workflows/python-release.yml (hunk 1 above)
Comment (antisaling) [reply .github/workflows/python-release.yml:6]: I'm not a fan of git tags myself, tbh. The alternative I found was to use something like
```sh
bash -o pipefail -c "poetry version | cut -d' ' -f2"
```
to get the version for two subsequent commits (e.g. this [version tagging action](https://github.com/salsify/action-detect-and-tag-new-version/blob/master/src/main.ts#L12) that [hypermodern-python cookiecutter](https://github.com/cjolowicz/cookiecutter-hypermodern-python/blob/main/%7B%7Bcookiecutter.project_name%7D%7D/.github/workflows/release.yml#L45) uses). wdyt?

Edit: typo in the command

Comment (tarilabs) [reply .github/workflows/python-release.yml:6]: Whatever works easiest, I'm just dubious in having separate tagging for Go/Python will be maintained?

We can definitely start with this setup and change later 👍 

(I'm assuming if one inadvertently tag as `py-v0.2.0` while having forgotten to update toml file: we just get an error from pypi; so we update the toml, and re-tag)

Comment (dtrifiro) [reply .github/workflows/python-release.yml:6]: The actual version uploaded to PyPi is indeed set by the `pyproject.toml`. I'd add a check to make sure that `pyproject.toml` version matches the git tag before running the release wortkflow.

Comment (tarilabs) [reply .github/workflows/python-release.yml:6]: that's a nice idea +1

Comment (antisaling) [reply .github/workflows/python-release.yml:6]: addressed on [+38-+42](https://github.com/opendatahub-io/model-registry/pull/230/files#diff-08d5d79624884efe962c68a256bee91079ad74acb3b22d61850797f5bf8aae04R38-R42), wdyt?

Comment (tarilabs) [reply .github/workflows/python-release.yml:6]: > addressed on [+38-+42](https://github.com/opendatahub-io/model-registry/pull/230/files#diff-08d5d79624884efe962c68a256bee91079ad74acb3b22d61850797f5bf8aae04R38-R42), wdyt?

I'm not expert in bash scripting, but the overall look of the command and the explanation of the command by LLM matches expectations :) so looks good to me, thank you!

**Review by tarilabs — DISMISSED** [review]

**Review by tarilabs — APPROVED** [review]

**Pull request merged**